### ISSUE TITLE
Suggestion: Using N-BEATS / N-HITS for Enhanced Sneaker Price Prediction

### ISSUE THREAD

**Issue by ManjuBodi** [sahithinandikula/Sneaker-price-prediction#43]
<html>
<body>
<p data-start="371" data-end="780"><strong data-start="371" data-end="383">Context:</strong><br data-start="383" data-end="386">
The current model uses an LSTM-based architecture to forecast sneaker resale prices based on historical time series data. While LSTM is a powerful tool for sequential modeling, recent advances in time series forecasting have shown that <strong data-start="622" data-end="673">pure feedforward models like N-BEATS and N-HITS</strong> can outperform RNN-based models, especially in terms of <strong data-start="730" data-end="779">accuracy, training stability, and scalability</strong>.</p>
<hr data-start="782" data-end="785">
<h3 data-start="787" data-end="810"> Proposed Change:</h3>
<p data-start="811" data-end="929">Migrate or experiment with <strong data-start="838" data-end="849">N-BEATS</strong> or <strong data-start="853" data-end="863">N-HITS</strong> as an alternative forecasting model for sneaker price prediction.</p>

<hr>

<h3>Why Consider N-BEATS / N-HITS?</h3>
<ul>
  <li><strong> Architecture:</strong><br>
    N-BEATS and N-HITS use pure feedforward networks (no recurrence), making them simpler and faster to train. In contrast, LSTM relies on sequential processing, which limits parallelization.</li>

  <li><strong> Forecasting Power:</strong><br>
    These models excel at both univariate and multivariate time series forecasting, often outperforming RNN-based models like LSTM, especially for longer horizons.</li>

  <li><strong> Training Stability:</strong><br>
    Gradient flow is more stable, with fewer issues like vanishing or exploding gradients, which are common in deeper LSTM setups.</li>

  <li><strong> Parallelization:</strong><br>
    N-BEATS/N-HITS architectures allow full parallelization across time steps, resulting in faster training and inference compared to LSTM’s step-wise nature.</li>

  <li><strong> Benchmark Performance:</strong><br>
    These models have achieved state-of-the-art results on public benchmarks such as M4 and M5, surpassing traditional LSTM and even hybrid models.</li>

  <li><strong> Interpretability:</strong><br>
    Especially in N-BEATS, the model structure allows decomposition into trend and seasonality blocks, providing a clearer understanding of the forecast components.</li>
</ul>

<hr>

</div></div>
<h3 data-start="1641" data-end="1673"> Benefits to This Project:</h3>
<ul data-start="1674" data-end="2134">
<li data-start="1674" data-end="1792">
<p data-start="1676" data-end="1792"><strong data-start="1676" data-end="1707">Improved Forecast Accuracy:</strong> N-BEATS has shown state-of-the-art performance in many real-world forecasting tasks.</p>
</li>
<li data-start="1793" data-end="1913">
<p data-start="1795" data-end="1913"><strong data-start="1795" data-end="1827">Faster Training &amp; Inference:</strong> Due to its non-recurrent nature, N-BEATS/N-HITS can utilize full parallelism on GPUs.</p>
</li>
<li data-start="1914" data-end="2036">
<p data-start="1916" data-end="2036"><strong data-start="1916" data-end="1949">Better Long-Term Forecasting:</strong> If sneaker trends change slowly or seasonally, N-BEATS models can better capture that.</p>
</li>
<li data-start="2037" data-end="2134">
<p data-start="2039" data-end="2134"><strong data-start="2039" data-end="2064">Simpler Architecture:</strong> Easier to maintain and tune than deep stacked LSTMs or hybrid models.</p></li></ul>
</body>
</html>

<hr>

P.S : N-BEATS and N-HiTS are two neural network models for time series forecasting. N-BEATS is a feed-forward network that uses a basis expansion approach and demonstrates strong performance without relying on traditional recurrent architectures. N-HiTS (Neural Hierarchical Time Series) is a more advanced extension of N-BEATS that improves long-term forecasting by employing multi-rate input sampling and hierarchical interpolation, allowing it to capture patterns at different time scales.
